src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs

```diff
@@ -428,6 +428,10 @@ private object ProcessValue(object obj, int depth)
             {
                 rv = obj;
             }
+            else if (obj is Newtonsoft.Json.Linq.JProperty)
+            {
+                rv = obj.ToString();
+            }
```

Comment:
Maybe change it to

```csharp
            else if (obj is Newtonsoft.Json.Linq.JObject jObject)
            {
                rv = jObject.ToObject<Dictionary<object,object>>();
            }
```

I tested that and it can support the following:

```powershell
$EgJObject = New-Object -TypeName Newtonsoft.Json.Linq.JObject
$EgJObject.Add("TestValue1", "123456")
$EgJObject.Add("TestValue2", "78910")
$EgJObject.Add("TestValue3", "99999")
$EgJObject.Add($null, "88888")
$EgJObject.Add("StringTest", "'StringToken'")
$EgJObject.Add("HashTest", '{ "innerkey": "innervalue" }')
$dict = @{}
$dict.Add('JObject', $EgJObject)
$dict.Add('StrObject', 'This is a string Object')
$properties = @{'DictObject' = $dict; 'RandomString' = 'A quick brown fox jumped over the lazy dog'}
$object = New-Object -TypeName psobject -Property $properties
$object | ConvertTo-Json
```

result
```json
{
  "DictObject": {
    "StrObject": "This is a string Object",
    "JObject": {
      "TestValue1": 123456,
      "TestValue2": 78910,
      "TestValue3": 99999,
      "": 88888,
      "StringTest": "StringToken",
      "HashTest": {
        "innerkey": "innervalue"
      }
    }
  },
  "RandomString": "A quick brown fox jumped over the lazy dog"
}
```

The null key value should on be an issue in the `ConvertFrom-Json` cmdlet. Since a null value key is valid JSON, we don't need to worry about it in this direction.

Reply:
Interesting. Newtonsoft.Json has large internal capabilities for serialization and deserialization - it would make sense to use them in future.